frontend/lib/src/components/widgets/CustomComponent/ComponentInstance.tsx

```diff
@@ -281,7 +281,7 @@ function ComponentInstance(props: Props): ReactElement {
     // Update the reference fields for the callback that we
     // passed to the componentRegistry
     onBackMsgRef.current = {
-      isReady: isReadyRef.current,
+      isReady: () => isReadyRef.current,
```

Comment:
nit: Could you please leave a comment for why we make this a callback. Probably a summary of your PR description aka: 
// isReady is a callback so that the caller is always getting the newest `isReadyRef.current` version 
or similar